### PR TITLE
group messages UI

### PR DESCRIPTION
**Pull request by slr**
group messages
group messages — new group creation
group messages — join group
RU_Chat group profile with active 'invite' button
RU_Chat group profile with active 'leave' button
RU_Chat group messages

* includes #297
* requires miguelfreitas/twister-core#346

TODO
- [x] show usernames in group messages
- [ ] ability to change group description
- [x] alias autocompletion for peers invitation form
- [ ] new messages counters in direct messages modals for every peer and group
- [ ] work with *nin*'s SASS
- [ ] mobile UI
- [ ] other

### CODE REVIEW

**Comment by slr** [comment]
group messages with names

**Comment by slr** [comment]
and in direct messages too.

File: js/twister_directmsg.js [hunk 1]
```diff
@@ -150,7 +156,8 @@ function directMessagesPopup() {
         .attr('title', polyglot.t('Mark all as read'))
         .on('click', function() {
             for (var user in _newDMsPerUser) {
-                _newDMsPerUser[user] = 0;
+                if (!user[0] === '*')
```
Comment (erqan) [inline js/twister_directmsg.js]: @slr  this doesn't work here, i think it should be `if (user[0] !== '*')`...

File: js/twister_newmsgs.js [hunk 2]
```diff
@@ -207,7 +225,18 @@ function getNewDMsCount() {
     var newDMs = 0;
 
     for (var user in _newDMsPerUser) {
-        if (_newDMsPerUser[user])
+        if (!user[0] === '*' && _newDMsPerUser[user])
```
Comment (erqan) [inline js/twister_newmsgs.js]: @slr same... this doesn't work here, i think it should be `if (user[0] !== '*')`

Comment (slr) [inline js/twister_newmsgs.js]: oops, my bad. you're right of course.

**Comment by miguelfreitas** [comment]
Hey, great job!!
Just returning from vacations and... merging :-)

**Comment by miguelfreitas** [comment]
I just don't know if we should give emphasis to the group alias (the *dfsdfsdf thing). It is a locally generated random string, so it is only meaningful if one wants to play with twisterd command line / rpc. 

For normal UI users it has no use, we should just present the group description instead.

Ok, i know... we may have two groups with the same name... but then it is the user who is confusing himself, no? ;-) I don't know how whatsapp UI behaves when two groups have the same name, for instance.

**Comment by slr** [comment]
speaking about description — what's with name and avatar? as I remember group profile based on ordinary one, can we set these things?

also can we send status message like invitation one when peer leaves group?

I'll continue to work on it next week.

**Comment by slr** [comment]
@miguelfreitas @erqan guys?

**Comment by slr** [comment]
+ issue with `rescandirectmsgs` — is it should be separate button (in (G)DM's window(s) or network section or in both) or we should detect peer invitation in DM group and rescan DMs automatically?

**Comment by miguelfreitas** [comment]
Maybe we should try triggering rescandirectmsgs automatically when a given user joins the group. It *shouldn't* cause any harm (duplicated messages are detected and not delivered again) unless some bug exists.

**Comment by miguelfreitas** [comment]
any group member may change description, but we have to think on how to implement avatar. perhaps sending an encrypted message to the group with avatar encoded.

**Comment by slr** [comment]
> ability to change group description

see miguelfreitas/twister-html#312

**Comment by slr** [comment]
> new messages counters in direct messages modals for every peer and group

see https://github.com/miguelfreitas/twister-html/commit/50827a6691829e50b592819ab22168f751e5c23e

**Pull request merged**